### PR TITLE
feat: Use caplog to test log messages

### PR DESCRIPTION
**Pull request by Glutexo**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] No Sensitive Data in this change?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

Logger asserts are error-prone, because they assert calls, not actual messages. The `caplog` fixture is Pytest-native way to test logs. This makes integration-level unit tests easier to add.

For the `caplog` fixture to work on Python 2.6 (RHEL 6), `pytest_catchlog` needs to be installed. On more recent version of Python, the fixture is included in Pytest.

`pytest_catchlog` caused PID file write tests to fail. The fixture itself calls `os.getpid`, interfering with the patch. Every log message makes a `getpid` call that is recorded in the patch. `assert_not_called` patches then stopped working.

Fixed by removing the `os.getpid` patch. The original function is safe to use. `test_write_pidfile_not_called` verifies several calls among which `write_to_disk` is the important one, not `getpid`.

Card IDs:

* [CCT-774](https://issues.redhat.com/browse/CCT-774) closed at the time as WONTDO
* [CCT-963](https://issues.redhat.com/browse/CCT-963) blocked by this patch
* [CCT-1084](https://issues.redhat.com/browse/CCT-1084) duplicate of the original card, now relevant

### CODE REVIEW

**Comment by Glutexo** [comment]
May I ask for a review, @m-horky? For more information on why this is here, see my [comment](https://issues.redhat.com/browse/CCT-1084?focusedId=26303878&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-26303878) in [CCT-1084](https://issues.redhat.com/browse/CCT-1084).

File: insights/tests/client/connection/test_test_connection.py [hunk 1]
```diff
@@ -224,55 +226,59 @@ def test_test_urls_raise(
...
+    http_method, request_function, insights_connection, caplog
 ):
     """The non-legacy URL subtest doesn't log any errors if all API calls succeed."""
     insights_connection.config.legacy_upload = False
 
-    with mock.patch(request_function):
-        insights_connection._test_urls(UPLOAD_URL, http_method)
+    with caplog.at_level(logging.ERROR, logger="insights.client.connection"):
+        with mock.patch(request_function):
+            insights_connection._test_urls(UPLOAD_URL, http_method)
```
[the remark this answers is not in the document]
Comment (Glutexo) [reply insights/tests/client/connection/test_test_connection.py:255]: Unfortunately, multiple `with` clauses are [not supported](https://docs.python.org/3.9/reference/compound_stmts.html#the-with-statement) in Python ≤ 2.7.

> Changed in version 3.1: Support for multiple context expressions.

The `patch` is there to mock a successful request. Without the patch, real HTTP requests would be made. Also, it was not introduced by this pull request.

File: insights/tests/client/connection/test_test_connection.py [hunk 2]
```diff
@@ -480,74 +484,80 @@ def test_legacy_test_urls_raise(
...
-    assert logger.error.mock_calls == [
-        mock.call(
-            "Could not successfully connect to: %s",
-            LEGACY_URL + LEGACY_URL_SUFFIXES[0],
+    assert caplog.record_tuples == [
+        (
+            "insights.client.connection",
+            logging.ERROR,
+            "Could not successfully connect to: %s"
+            % (LEGACY_URL + LEGACY_URL_SUFFIXES[0],),
```
[the remark this answers is not in the document]
Comment (Glutexo) [reply insights/tests/client/connection/test_test_connection.py:534]: Nice catch!

File: insights/tests/client/connection/test_test_connection.py [hunk 3]
```diff
@@ -657,42 +667,46 @@ def test_test_connection_test_urls_fail(test_urls, exception, insights_connectio
...
-    except UnexpectedException:
-        pass
+    with caplog.at_level(logging.ERROR, logger="insights.client.connection"):
+        try:
+            insights_connection.test_connection()
+        except UnexpectedException:
+            pass
 
-    logger.error.assert_not_called()
+        assert not caplog.record_tuples
```
[the remark this answers is not in the document]
Comment (Glutexo) [reply insights/tests/client/connection/test_test_connection.py:681]: You’re right!

**Review by Glutexo — COMMENTED** [review]
Thank you, @pkoprda, for a precise review! ❤️ I fixed the issues you spotted and looked through the code for other instances. You spotted them all. Good job!

File: insights/tests/client/connection/test_test_connection.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkoprda) [reply insights/tests/client/connection/test_test_connection.py:254]: Okay, then it's fine

File: insights/tests/client/connection/test_test_connection.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (pkoprda) [reply insights/tests/client/connection/test_test_connection.py:679]: Still applies

[the remark this answers is not in the document]
Comment (Glutexo) [reply insights/tests/client/connection/test_test_connection.py]: 🤦🏻‍♂️  Fixed.

**Comment by pkoprda** [comment]
@Glutexo looks good, just squash the commits and then I will approve it

**Comment by Glutexo** [comment]
Sure thing! Squashed. @pkoprda

**Review by pkoprda — APPROVED** [review]
Thank you @Glutexo, LGTM! The failing test is not related to this change.

@xiangce can we merge this?

**Comment by Glutexo** [comment]
@pkoprda You were faster. 😃 I wanted to do that too.

**Pull request merged**